File: PS/algo/src/Leetcode/easy/BinarySearch.java
package Leetcode.easy;

public class BinarySearch {
    public int search(int[] nums, int target) {
        int start = 0;
        int end = nums.length - 1;
        int answer = -1;
        while (start <= end) {
            int point = (start + end) / 2;
            if (nums[point] < target) {
                start = point+1;
            } else if (nums[point] > target) {
                end = point-1;
            } else if (nums[point] == target) {
                answer = point;
                break;
            }
        }
        return answer;
    }
}

File: src/main/kotlin/BinarySearch.kt
class BinarySearch {
    fun search(nums: IntArray, target: Int): Int {
        var answer = -1
        var start = 0
        var end =nums.size-1
        while (start<=end){
            val mid  = (start+end)/2
            if(nums[mid]<target)start =mid+1
            else if(nums[mid]>target)end = mid-1
            else return mid
        }
        return answer
    }
}